[Rayan Wong] iP

Rayan-Wong opened the pull request.

Rayan-Wong committed:
Level-0

Rayan-Wong committed:
Level-1

Added CLI parsing of commands
in Cli.java. Currently, it just prints out what was typed into, apart from "bye" which stops the program

Rayan-Wong committed:
Level-2

Added add and list all tasks functionality. Additionally, created TaskService interface, TaskService implementation and TaskServiceRepo

Rayan-Wong committed:
Align switch-case block in CLI to follow code style

Rayan-Wong committed:
Added mark and unmarking of tasks as completed.

Tasks now have a done attribute which is set via markDone(status). TaskService supplies the status via CLI. Generic index verifier + aligner created to use across whole TaskServiceRepo class, also throwing IndexOutOfBoundsException if an invalid index is supplied.

Rayan-Wong committed:
Modified add task behaviour

Tasks are now added by simply typing the task name, instead of calling the keyword add

Mack34021 commented on README.md:
```diff
@@ -1,6 +1,6 @@
-# Duke project template
+# Bob project template
```
You updated the readme, nice, not everyone remembered that.

Mack34021 commented on src/main/java/Cli.java:
```diff
@@ -0,0 +1,67 @@
+                break;
+
+            case "mark":
+                if (cmd.length <= 1) {
+                    System.out.println("No task index specified!");
+                } else {
+                    int index = Integer.parseInt(cmd[1]);
+                    try {
+                        service.completeTask(repo, index);
+                        String task = service.fetchTask(repo, index);
+                        System.out.println("I've marked the following task as done: " + task);
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.println("Invalid index: " + index);
+                        continue;
+                    }
+                }
+                break;
+
```
Not an actual coding standard violation, but maybe consider extracting switch cases into a method to make the switch more readable e.g.

case "mark":
    markTask(index);
    break;
case "unmark"
    unMarkTask(index);
    break;
etc.

Mack34021 commented on src/main/java/Cli.java:
```diff
@@ -0,0 +1,67 @@
+        Scanner sc = new Scanner(System.in);
+        // Continuously loop CLI until "bye" command is entered
+        while (true) {
+            // strip excess spaces
```
In terms of comments, some are capitalised others not. Might be better to be consistent across entire project.

Mack34021 commented on src/main/java/TaskServiceRepo.java:
```diff
@@ -0,0 +1,35 @@
+        return new ArrayList<Task>(repo);
+    }
+
+    public void mark(int index, boolean status) {
+        index = validateIndex(index);
+        repo.get(index).markDone(status);
+    }
+
+    public Task fetchTask(int index) {
+        index = validateIndex(index);
+        return repo.get(index);
+    }
```
Method naming: Some methods names are more describtive than others "mark()" vs "fetchTask()". Potentially consider being more consistent with the naming i.e. "mark()" -> "markTask()" or vice versa

Mack34021 reviewed (commented):
Looking pretty good! Very nice class decomposition. I didn't actually find any blatant coding standards violations.

Kurokishi592 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,5 @@
+    public Event(String desc) {
+        super(desc);
+    }
+}
```
Great job on keeping the code clean. However, this class is simply calling the superclass constructor, perhaps you can consider augmentation by adding new functions, or overriding/overloading superclass functions. Same for Deadline and Todo classes too

Kurokishi592 commented on src/main/java/TaskServiceRepo.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.ArrayList;
+
+// Class acts as a "dumb" repo: processing is done at service
```
Have you considered using the proper Javadoc commenting style? You may refer to https://se-education.org/guides/conventions/java/basic.html#comments

Kurokishi592 commented on src/main/java/Cli.java:
```diff
@@ -0,0 +1,67 @@
+                Task task = new Task(task_name);
+                service.addTask(repo, task);
+                System.out.println("added: " + task_name);
+            }
```
Good job on the indentations for your switch statement!

Kurokishi592 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+    public boolean getDone() {
+        return done;
+    }
+}
```
Good use of K&R style brackets (aka [Egyptian style]) !

Kurokishi592 commented on src/main/java/TaskServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+    public String fetchTask(TaskServiceRepo repo, int index) {
+        return repo.fetchTask(index).getDesc();
+    }
+}
```
Nice to see that you added @Override to indicate the overriding implementations!

Kurokishi592 reviewed (commented):
Overall, your code followed the Java coding standards well. LGTM!

Rayan-Wong committed:
Added todo, event, deadline

Using keywords todo, event and deadline respectively, you can now add them. On success, the task status is now also printed including how many tasks are in the list. Tasks refer to todo, event or deadline. Added toString() methods for all of such classes. Without using a keyword, Bob now rejects the command.

Rayan-Wong committed:
Refactored string array joining in Cli to join(). Added checks for invalid from, to and by args.

Rayan-Wong committed:
Improved test to accomodate for happy paths and common edge cases (no task name, using list command when its empty, no /by arg for deadline, no /from and /to for event)
Added handling of when list command is used if there are no tasks

Rayan-Wong committed:
Moved all java files into a single package called bob, updated test scripts to reflect changes

todo: Package all java files into relevant packages

Rayan-Wong committed:
Placed classes etc in appropriate packages, fixed test script to work with current packaging, fixed README.md

Rayan-Wong committed:
Merge pull request #1

branch-A-Packages

Packaged classes etc into proper packages

Rayan-Wong committed:
Included proper formatting use for error messages

Rayan-Wong committed:
Merge pull request #2

Included proper formatting use for error messages

Rayan-Wong committed:
Added app specific exceptions for repo indexing exceptions, modified exception messages, refactored command syntaxes, used exceptions for all invalid inputs except only if user only use one word for invalid commands, updated test, improved readability

Rayan-Wong committed:
Merge pull request #3

Added app specific exceptions for repo indexing exceptions, modified exception messages, refactored command syntaxes, used exceptions for all invalid inputs except only if user only use one word for invalid commands, updated test, improved readability

syCHEN1645 commented on src/main/java/bob/adapter/Cli.java:
```diff
@@ -0,0 +1,157 @@
+import java.util.List;
+
+public class Cli {
+    public static final String TODO_SYNTAX = "todo <task name>";
```
Good job managing input syntax!

syCHEN1645 commented on src/main/java/bob/adapter/Cli.java:
```diff
@@ -0,0 +1,157 @@
+            // strip excess spaces
+            String[] cmd = sc.nextLine().replaceAll("\\s+", " ").split(" ");
+            switch (cmd[0]) {
+            case "bye":
```
Maybe consider replacing magic literals with constant strings?

syCHEN1645 commented on src/main/java/bob/adapter/Cli.java:
```diff
@@ -0,0 +1,157 @@
+            default:
+                // add task functionality
+                System.out.println("Sorry, please type a valid input!");
+            }
```
Consistent level of abstraction, good job!

syCHEN1645 commented on src/main/java/bob/models/Event.java:
```diff
@@ -0,0 +1,22 @@
+package bob.models;
+
+public class Event extends Task {
+    String from;
```
How about using nouns to name these variables?

syCHEN1645 commented on src/main/java/bob/models/Task.java:
```diff
@@ -0,0 +1,28 @@
+package bob.models;
+
+public class Task {
+    private final String desc;
```
I am not sure if this a common abbreviation? Please ignore if it is.

syCHEN1645 reviewed (commented):
Very neat and structured code, keep it up!

Rayan-Wong committed:
Added delete functionality for tasks, modified fetchtask() to return Task type, updated input test for coverage against edge case

Rayan-Wong committed:
Updated fetch() and remove() to reflect repo is more generic

Rayan-Wong committed:
Added file-saving capabilities

Rayan-Wong committed:
Merge pull request #5

Added delete functionality for tasks, modified fetchtask() to return Task type, updated input test for coverage against edge case

Rayan-Wong committed:
Merge branch 'branch-Level-7' of https://github.com/Rayan-Wong/ip into branch-Level-7

# Conflicts:
#	src/main/java/bob/adapter/Cli.java

Rayan-Wong committed:
Merge pull request #6 from Rayan-Wong/branch-Level-7

Added file-saving capabilities

Rayan-Wong committed:
Fixed file directory set to relative instead of literal <root>/data (sorry to whoever got affected :()

Rayan-Wong committed:
Merge pull request #7

Fixed file directory set to relative instead of literal <root>/data (sorry to whoever got affected :()

Rayan-Wong committed:
Fixed file directory set to relative instead of literal <root>/data (sorry to whoever got affected :(), added done status saving, added saving Bob state when user leaves (and quits)

Rayan-Wong committed:
Merge pull request #8

Fixed file directory set to relative instead of literal <root>/data (sorry to whoever got affected :(), added done status saving, added saving Bob state when user leaves (and quits)

Rayan-Wong committed:
Fixed META-INF

Rayan-Wong committed:
Merge pull request #9

Fixed META-INF

Rayan-Wong committed:
Added find task from substring functionality, moved deserialising to model classes instead of FileHelperImpl, updated tests

Rayan-Wong committed:
Merge pull request #10

Added find task from substring functionality, moved deserialising to model classes instead of FileHelperImpl, updated tests

Rayan-Wong committed:
Added JavaDoc comments to all classes, public methods and non-trivial private methods

Rayan-Wong committed:
Merge pull request #11

Added JavaDoc comments to all classes, public methods and non-trivial private methods

Rayan-Wong committed:
Added datetime functionality for Events and Deadline, including checking of syntax. RChanged base class Task's variables to protected.

Rayan-Wong committed:
Improved cmd syntax checking, fixed datetime formatting from getDesc() for event, updated tests

Rayan-Wong committed:
Added find by date functionality for deadlines and events, updated tests

Rayan-Wong committed:
Merge pull request #12

branch-Level-8

Rayan-Wong committed:
Updated docs/README.md for Github Pages, fixed syntax to display date for findbydate, updated tests, added Gemfile to allow local deployment of Github pages for README

Rayan-Wong committed:
Merge pull request #13

Updated docs/README.md for Github Pages, fixed syntax to display date for findbydate, updated tests, added Gemfile to allow local deployment of Github pages for README
